XenModel/PoolJoinRules.cs

```diff
@@ -382,6 +388,27 @@ private static bool DifferentServerVersion(Host slave, Host master)
                 slave.ProductBrand != master.ProductBrand;
         }
 
+        private static bool DifferentPatches(Host slave, Host master)
+        {
+            if (Helpers.ElyOrGreater(master))
+            {
+                var masterUpdates = master.AppliedUpdates();
+                var slaveUpdates = slave.AppliedUpdates();
+                if (masterUpdates.Except(slaveUpdates, new UpdateUuidEqualityComparer()).Any() ||

```

Comment:
It would be more efficient to check that the length of the two Lists is the same, and that the set difference in one direction is empty. That way, you only have to calculate at most one set difference, rather than two; and in most cases of inequality, zero.